tuning/sft_trainer.py

```diff
@@ -213,28 +215,54 @@ def train(
         fusedops_kernels_config,
     ).get_framework()
 
-    model_loader = AutoModelForCausalLM.from_pretrained
-    if framework is not None and framework.requires_custom_loading:
-        model_loader = framework.model_loader  # drop-in new loader
+    # option to set multimodal var here
     model_load_time = time.time()
-    model = model_loader(
-        model_args.model_name_or_path,
-        cache_dir=train_args.cache_dir,
-        torch_dtype=get_torch_dtype(model_args.torch_dtype),
-        attn_implementation="flash_attention_2" if model_args.use_flash_attn else None,
-    )
+    processor = None
+    try:
+        # try to load vision model
+        model_loader = AutoModelForVision2Seq.from_pretrained
+        model = model_loader(
+            model_args.model_name_or_path,
+            cache_dir=train_args.cache_dir,
+            torch_dtype=get_torch_dtype(model_args.torch_dtype),
+            attn_implementation="flash_attention_2"
```

Comment:
is attn_implementation="flash_attention_2" working for you @anhuong with the llama vision model?

when I try the standalone SFT script you gave me , it is says not supported for mllama architecture, also in https://github.com/huggingface/transformers/issues/35634

Reply:
It doesn't work for me either for llama thus I set `"use_flash_attn": false,` for llama but flash attention does work for granite and llava models. Writing up docs explaining usage